applications/wikipedia/main.py

```diff
@@ -153,22 +160,35 @@ def embed_dataset(down_scale: float = 0.005, batch_size: int = 32):
     print(f"Working with {sample_size} rows")
 
     text_chunks = generate_chunks_from_dataset(subset, chunk_size=400)
-    batches = generate_batches(
-        text_chunks, batch_size=batch_size
-    )  # 32 is the max batch size of the model
+    batches = generate_batches(text_chunks, batch_size=batch_size)
 
     start = time.perf_counter()
-    materialized_batchs = list(batches)
+    materialized_batches = list(batches)
     print(
-        f"Materialized {len(materialized_batchs)} batches in {time.perf_counter()-start:.2f} seconds"
+        f"Materialized {len(materialized_batches)} batches in {time.perf_counter()-start:.2f} seconds"
     )
 
     start = time.perf_counter()
     counter = 0
-    for n_chars in model.embed.map(materialized_batchs, order_outputs=False):
-        counter += n_chars
+    texts = []
+    embeddings = []
+    for text_chunk_batch, computed_embedding in zip(
+        materialized_batches, model.embed.map(materialized_batches, order_outputs=False)
+    ):
+        counter += sum(map(len, text_chunk_batch))
+        texts.extend(text_chunk_batch)
+        embeddings.extend(computed_embedding)
+
     end = time.perf_counter()
 
+    if PUSH_TO_HUB:
```

Comment:
This looks promising 

https://github.com/huggingface/huggingface_hub/issues/1085

Do you think it might be wise then to write the saved batch into a file and then save it to the volume before dispatching another container to read and to the upload